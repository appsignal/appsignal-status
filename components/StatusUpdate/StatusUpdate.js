import PropTypes from "prop-types";
import dayjs from "dayjs";
import advancedFormat from "dayjs/plugin/advancedFormat";
dayjs.extend(advancedFormat);

import StatusIcon from "../StatusIcon/StatusIcon";

const StatusUpdate = ({ update }) => {
  return (
    <div
      className="flex bg-white shadow-sm rounded py-5 px-6"
      data-testid="statusUpdate"
    >
      <div className="mr-4 my-0.5">
        <StatusIcon status={update.state} />
      </div>
      <div className="">
        <h3 className="c_h-heading mb-1">{update.title}</h3>
        {update.description && (
          <p className="text-gray-700">{update.description}</p>
        )}
        <p className="text-gray-600 text-ms mt-2">
          {dayjs(update.time).format("MMM. Do H:m")}
        </p>
      </div>
    </div>
  );
};

StatusUpdate.propTypes = {
  update: PropTypes.shape({
    state: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    description: PropTypes.string,
    time: PropTypes.string.isRequired,
  }).isRequired,
};

export default StatusUpdate;
