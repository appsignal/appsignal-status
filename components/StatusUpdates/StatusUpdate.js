import PropTypes from "prop-types";

import StatusIcon from "../StatusIcon/StatusIcon";

const StatusUpdate = ({ status, title, description, time }) => {
  return (
    <div className="flex bg-white shadow-sm rounded py-5 px-6">
      <div className="mr-4 my-0.5">
        <StatusIcon status={status} />
      </div>
      <div className="">
        <h3 className="c_h-heading mb-1">{title}</h3>
        {description && <p className="text-gray-700">{description}</p>}
        <p className="text-gray-600 text-ms mt-2">{time}</p>
      </div>
    </div>
  );
};

StatusUpdate.propTypes = {
  status: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  description: PropTypes.string,
  time: PropTypes.string.isRequired,
};

export default StatusUpdate;
