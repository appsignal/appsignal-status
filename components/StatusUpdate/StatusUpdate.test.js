import { render, screen } from "@testing-library/react";

import statusPageMock from "../../mocks/status_pages/appsignal.json";

import StatusUpdate from "./StatusUpdate";

const build = (props = {}) => {
  return render(<StatusUpdate update={statusPageMock.updates[0]} {...props} />);
};

describe("StatusUpdate", () => {
  test("renders without errors", () => {
    const { container } = build();
    expect(container).toMatchSnapshot();
  });

  test("it shows title description and formatted time", () => {
    build({ update: statusPageMock.updates[1] });
    expect(screen.getByRole("img", { hidden: true }).classList).toContain(
      "fa-check"
    );
    expect(screen.getByText("All Fixed again")).toBeInTheDocument();
    expect(screen.getByText("All good")).toBeInTheDocument();
    expect(screen.getByText("Sep. 1st")).toBeInTheDocument();
  });
});
