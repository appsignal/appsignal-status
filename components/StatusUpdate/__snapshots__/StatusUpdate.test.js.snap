// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusUpdate renders without errors 1`] = `
<div>
  <div
    class="flex bg-white shadow-sm rounded py-5 px-6"
    data-testid="statusUpdate"
  >
    <div
      class="mr-4 my-0.5"
    >
      <span
        class="flex items-center justify-center h-4 w-4 rounded-full bg-red-500 text-white"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-exclamation fa-w-6 fa-fw "
          data-icon="exclamation"
          data-prefix="fas"
          focusable="false"
          role="img"
          style="font-size: 8px;"
          viewBox="0 0 192 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M176 432c0 44.112-35.888 80-80 80s-80-35.888-80-80 35.888-80 80-80 80 35.888 80 80zM25.26 25.199l13.6 272C39.499 309.972 50.041 320 62.83 320h66.34c12.789 0 23.331-10.028 23.97-22.801l13.6-272C167.425 11.49 156.496 0 142.77 0H49.23C35.504 0 24.575 11.49 25.26 25.199z"
            fill="currentColor"
          />
        </svg>
      </span>
    </div>
    <div
      class=""
    >
      <h3
        class="c_h-heading mb-1"
      >
        Some update
      </h3>
      <div
        class="text-gray-700"
      >
        <p>
          Some description
        </p>
      </div>
      <p
        class="text-gray-600 text-ms mt-2"
      >
        Aug. 30th 08:43
      </p>
    </div>
  </div>
</div>
`;
